### PR TITLE
Use hash_newtype macro instead of the hash256 module

### PR DESCRIPTION
**Pull request by tcharding**
We currently implement the `hash256::Hash` type manually however there is a macro in `bitcoin::hashes` that can do that for us.

For context, the `hash256::Hash` type exists solely to enable the type to display forwards since `hashes::sha256d` displays backwards. The `hash_newtype` macro takes an argument to configure display.

### CODE REVIEW

**Comment by sanket1729** [comment]
can you cherry-pick this commit https://github.com/rust-bitcoin/rust-miniscript/pull/516/commits/62219d482ca05f3a969d18bb15445e084d0102d1 from #516? That should Fix CI

**Comment by tcharding** [comment]
Done! No other changes.

**Review by apoelstra — APPROVED** [review]
ACK 0b482cd755e3d7c693fffe54849f31204269d45a

File: src/miniscript/mod.rs [hunk 1]
```diff
@@ -459,6 +458,14 @@ impl_from_str!(
 
 serde_string_impl_pk!(Miniscript, "a miniscript", Ctx; ScriptContext);
 
+/// Provides a Double SHA256 `Hash` type that displays forwards.
+pub mod hash256 {
+    use bitcoin::hashes::{hash_newtype, sha256d};
+
+    #[rustfmt::skip]
+    hash_newtype!(Hash, sha256d::Hash, 32, doc = "A bitcoin block hash.", false);
```
Comment (sanket1729) [inline src/miniscript/mod.rs:466]: Merging this now to fix the CI. But we should update the doc-comment to reflect that this is just double-hashed sha256::Hash with display forward and not a bitcoin block hash.

**Review by sanket1729 — APPROVED** [review]
ACK 0b482cd755e3d7c693fffe54849f31204269d45a

**Pull request merged**